### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# ❓What kind of change does this PR introduce?

- [x] Add Login Functionality.
- [x] Add Signup Functionality. 

[Login](https://valo-tube-git-feature-authentication-kevin-solomon.vercel.app/login)
[Signup](https://valo-tube-git-feature-authentication-kevin-solomon.vercel.app/signup)

### CODE REVIEW

**Review by starver20 — APPROVED** [review]
Signup and login functionalities are working all good.
- You can give `cursor: pointer;` for the sidebar items and also the navbar items.
- You can increase the font-size of login and signup button a bit.

File: src/pages/public/Login/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        user,
+        authDispatch
+      );
+      if (response === 200) {
+        navigate('/', { replace: true });
+      } else if (response === 401) {
+        alert('Credentials are wrong');
+      } else if (response === 404) {
+        alert('You need to sign up, coz you are not registered');
+      } else {
```
Comment (AtulPant2704) [inline src/pages/public/Login/Login.jsx:25]: You don't need to check for multiple status, you can just throw an error and in catch(error) you can console.log all types of errors.

**Review by AtulPant2704 — APPROVED** [review]
Rest all looks good, functionalities are also working fine.

File: src/pages/public/Login/Login.jsx (hunk 1 above)
Comment (Kevin-Solomon) [reply src/pages/public/Login/Login.jsx:25]: But the catch block runs only when there is a network error, right?.

**Pull request merged**